File: Sport_football/football_tools/app/Http/Controllers/shop/shopController.php
<?php

namespace App\Http\Controllers\shop;

use App\Http\Controllers\Controller;
use Illuminate\Http\Request;
use App\topic;
use App\category;
use App\product;

class shopController extends Controller
{
    public function showProduct(Request $request)
    {
       
            $products = product::where('status', 0)
                ->inRandomOrder()
                ->limit(9)
                ->get();
        

        $max_price = product::where('status', 0)
            ->max('price');
        $min_price = product::where('status', 0)
            ->min('price');

        $topics = topic::get();
        $stt = 0;
        return view('shop.pages.shop', [
            'topics' => $topics,
            'stt' => $stt,
            'products' => $products,
            'max_price' => $max_price,
            'min_price' => $min_price
        ]);
    }

    public function productList(Request $request, $id_category)
    {
        if (isset($request->id_category)) {
            $products = product::where('id_category', $request->id_category)
                ->where('status', 0)
                ->get();
        }


        $output = null;

        if (count($products) > 0) {
            //  dd('chay vao day');
            $price_sale = 0;
            foreach ($products as $item) {
                $price_sale = $item->price - ($item->price * ($item->sale / 100));
                $a = $item->image;
                $output .= '  <div class="col-lg-4 col-md-6">
                <div class="product__item">
                    <div class="product__item__pic set-bg" data-setbg="' . $a . '" style="background-image: url(' . $a . ');">';

                if (isset($item->sale)) {
                    $output .= '<div class="label sale">Sale</div>';
                }

                $output .= '<ul class="product__hover">
                            <li><a href="' . $a . '" class="image-popup"><i class="fa fa-retweet"></i></a></li>
                            <li><a href="#"><i class="far fa-heart"></i></a></li>
                            <li><a href="#"><i class="fas fa-shopping-bag"></i></a></li>
                        </ul>
                    </div>
                    <div class="product__item__text">
                        <h6><a href="' . route('home') . '">' . $item->name_product . '</a></h6>
                        <div class="rating">
                            <i class="fa fa-star"></i>
                            <i class="fa fa-star"></i>
                            <i class="fa fa-star"></i>
                            <i class="fa fa-star"></i>
                            <i class="fa fa-star"></i>
                        </div>';
                if (isset($item->sale)) {
                    $output .=  '<div class="product__price">đ ' . number_format($price_sale) . '<span>' . number_format($item->price) . '</span></div>';
                } else {
                    $output .=  '<div class="product__price">đ ' . number_format($item->price) . '</div>';
                }
                $output .= '</div></div></div>';
            }
        } else {
            $output = ' <h2 style="margin-left: 200px;">No products</h2>';
        }
        echo $output;
    }
}

File: Hello/src/main/java/com/hello/controller/shopController.java
package com.hello.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.servlet.ModelAndView;

import com.hello.DAO.category_dao;
import com.hello.DAO.product_dao;
import com.hello.DAO.topic_dao;
import com.hello.entity.category;
import com.hello.entity.product;
import com.hello.service.product_service;

@Controller
@RequestMapping("/shop")
public class shopController {

	@Autowired
	topic_dao topic_dao;
	@Autowired
	category_dao category_dao;
	@Autowired
	product_dao product_dao;
	@Autowired
	product_service product_service;

	@GetMapping("")
	@Transactional
	public ModelAndView shop() {
		ModelAndView modelAndView = new ModelAndView("football/shop");
		modelAndView.addObject("list_topic", topic_dao.list_topic());
		modelAndView.addObject("list_category", category_dao.list_category());
		modelAndView.addObject("list_product",product_service.shuffleListProduct());
		return modelAndView;
	}

	@GetMapping(path = "/", produces = "text/plain; charset=utf-8")
	@ResponseBody
	public String shopCategory(@RequestParam int id_category) {
		String htmlString = "";
		for (product product : product_dao.list_product()) {
			if (product.getCategory().getId_category() == id_category) {
				htmlString += "<div class=\"col-lg-4 col-md-6\">\r\n"
						+ "					<div class=\"product__item\">\r\n"
						+ "						<div class=\"product__item__pic set-bg\"\r\n"
						+ "							data-setbg=\"" + product.getImage()
						+ "\" style=\"background-image: url(" + product.getImage() + ");\">\r\n"
						+ "							<div class=\"label new\">New</div>\r\n"
						+ "<ul class=\"product__hover\">\r\n" + 
						"								<li><a href=\""+product.getImage()+"\" class=\"image-popup\"><i\r\n" + 
						"										class=\"fa fa-retweet\"></i></a></li>\r\n" + 
						"								<li><a href=\"#\"><i class=\"far fa-heart\"></i></a></li>\r\n" + 
						"								<li><a href=\"#\"><i class=\"fas fa-shopping-bag\"></i></a></li>\r\n" + 
						"							</ul>" + "						</div>\r\n"
						+ "						<div class=\"product__item__text\">\r\n"
						+ "							<h6>\r\n" + "								<a href=\"#\">"
						+ product.getName_product() + "</a>\r\n" + "							</h6>\r\n"
						+ "							<div class=\"rating\">\r\n"
						+ "								<i class=\"fa fa-star\"></i> <i class=\"fa fa-star\"></i> <i\r\n"
						+ "									class=\"fa fa-star\"></i> <i class=\"fa fa-star\"></i> <i\r\n"
						+ "									class=\"fa fa-star\"></i>\r\n"
						+ "							</div>\r\n"
						+ "							<div class=\"product__price\">"+product.getPrice()+"</div>\r\n"
						+ "						</div>\r\n" + "					</div>\r\n" + "				</div>";
			}
		}
		if (htmlString.equals("")) {
			htmlString = "Not product";
		}
		return htmlString;

	}

}
